### PR TITLE
Add keyword arguments to draw.arc

### PR DESCRIPTION
**Pull request by charlesej**
Overview of changes:
- Added keyword arguments to `draw.arc`
- Added/updated some exception messages for `draw.arc`
- Added tests to check `draw.arc` args/kwargs
- Fixed test variable name
- Updated draw documentation

System details:
- os: windows 10 (64bit)
- python: 3.7.2 (64bit) and 2.7.10 (64bit)
- pygame: 2.0.0.dev0 (SDL: 1.2.15) at d84a08a31f7682dfd704f8d46dfd578031d742cd

Resolves sub-item `pygame.draw.arc` of item "Add keyword arguments to the draw functions..." of #896.

### CODE REVIEW

File: docs/reST/ref/draw.rst [hunk 1]
```diff
@@ -153,13 +153,57 @@ object around the draw calls (see :func:`pygame.Surface.lock` and
 
 .. function:: arc
 
-   | :sl:`draw a partial section of an ellipse`
-   | :sg:`arc(Surface, color, Rect, start_angle, stop_angle, width=1) -> Rect`
+   | :sl:`draw an elliptical arc`
+   | :sg:`arc(surface=Surface, color=Color, rect=Rect, start_angle=radians, stop_angle=radians) -> Rect`
```
Comment (dlon) [inline docs/reST/ref/draw.rst]: Shouldn't `=` be used for default values?

Comment (charlesej) [reply docs/reST/ref/draw.rst]: I've been using it to indicate named parameters as well. Should it be like this instead?
`arc(surface, color, rect, start_angle, stop_angle) -> Rect`
`arc(surface, color, rect, start_angle, stop_angle, width=1) -> Rect`

Comment (dlon) [reply docs/reST/ref/draw.rst]: IMO, that's better. Though I wonder if it would be useful to include some of the RST type info in the docstring somehow?

Comment (charlesej) [reply docs/reST/ref/draw.rst]: I've made the requested changes. I also opened issue #1020 to track this for the other shape functions.

Regarding the type info, currently the `:param:` and `:type:` directives are used to indicate any type information. Also there is issue #839 which is working on a general type strategy for pygame.

**Pull request merged**